### PR TITLE
Enable support for SSH private key distinct from the bastion private …

### PR DESCRIPTION
**Pull request by vikramraodp**
…key to connect to the host

* Add `host_private_key` to the resource schema
* Update function`resourceResourceCreate` to check & use the `host_private_key` when provided
* Update function `resourceResourceUpdate` to check & use the `host_private_key` when provided
* Update file `docs/resources/resource.md` to include the `host_private_key` in the example

### CODE REVIEW

File: ssh/resource_resource.go [hunk 1]
```diff
@@ -50,6 +51,12 @@ func resourceResource() *schema.Resource {
 				Sensitive:    true,
 				RequiredWith: []string{"user"},
 			},
+			"host_private_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				RequiredWith: []string{"user"},
```
Comment (loafoe) [inline ssh/resource_resource.go:58]: Since it's optional, should not use `RequiredWith`

**Review by loafoe — CHANGES_REQUESTED** [review]
Awesome feature! Just one item, once addressed will merge and release new version. Thanks!

File: ssh/resource_resource.go (hunk 1 above)
Comment (vikramraodp) [reply ssh/resource_resource.go:58]: Ah! I missed that indeed. I'll fix it & make another commit. Thanks Andy.

**Review by loafoe — APPROVED** [review]

**Comment by loafoe** [comment]
@vikramraodp release `v0.2.1` is available with this change from the registry, thx again!

**Pull request merged**